### PR TITLE
Add document describing how to disable tests

### PR DESCRIPTION
**Pull request by BruceForstall**
Update libraries "filtering tests" doc with information and examples for
using `SkipOnCoreClrAttribute`

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet/runtime-infrastructure @danmoseley I wrote a document on how to disable tests, and added to the libraries test filtering document information on using `SkipOnCoreClr` (as well as adding various examples). PTAL.

**Review by trylek — APPROVED** [review]
Beautiful, thanks Bruce for putting this together!

**Comment by jkoritzinsky** [comment]
You should note that all of the attributes used in the libraries tests are also supported in the runtime tests when using xunit attributes.

File: docs/workflow/testing/libraries/filtering-tests.md [hunk 1]
```diff
@@ -176,3 +234,67 @@ dotnet build <csproj_file> /t:Test /p:TargetOS=windows
 ```sh
 dotnet build <csproj_file> /t:Test /p:TargetOS=OSX /p:WithCategories="OuterLoop;failing""
 ```
+
+## SkipOnCoreClrAttribute
```
Comment (jkotas) [inline docs/workflow/testing/libraries/filtering-tests.md:238]: We also `SkipOnMonoAttribute`. 

On average, there are more tests disabled on Mono than on CoreCLR, so `SkipOnMonoAttribute` is more common.

File: docs/workflow/testing/libraries/filtering-tests.md [hunk 2]
```diff
@@ -176,3 +234,67 @@ dotnet build <csproj_file> /t:Test /p:TargetOS=windows
...
+This attribute is used to disable a test under specific conditions, only when run with CoreCLR (it doesn't affect tests run with Mono). Typically, this is when there is a failure in a particular test run configuration, such as under GCStress or JitStress.
```
Comment (jkotas) [inline docs/workflow/testing/libraries/filtering-tests.md:239]: I believe that `SkipOn...` attributes are meant to be used for permanent disable, without an ActiveIssue.

File: docs/workflow/testing/libraries/filtering-tests.md [hunk 3]
```diff
@@ -176,3 +234,67 @@ dotnet build <csproj_file> /t:Test /p:TargetOS=windows
...
+
+```cs
+[SkipOnCoreClr(string reason, TestPlatforms testPlatforms)]
+```
+Example:
+```cs
+[SkipOnCoreClr("Long running tests: https://github.com/dotnet/runtime/issues/11980", TestPlatforms.Linux)
+```
+
+**Disable for specific test mode:**
```
Comment (danmoseley) [inline docs/workflow/testing/libraries/filtering-tests.md:265]: Is there a way to disable a test for any "slow" mode? that might be a debug runtime, or some stress mode. Not sure whether that's a scenario.

File: docs/workflow/ci/disabling-tests.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+Tests are either disabled permanently or temporarily:
+- Permanently: the test is never expected to run in a certain configuration, due to the design of the test or product.
+- Temporarily: the test is failing, and needs to be disabled so a test run isn't continuously made "noisy" by the
+existence of a failing test in the results. These tests are expected to be re-enabled when a bug is fixed or
+feature is implemented.
+
+## Runtime or libraries?
+
+There are two main sets of tests in the repo: runtime tests in the src/tests tree, and libraries tests which
```
Comment (danmoseley) [inline docs/workflow/ci/disabling-tests.md:15]: are there any CLR specific or Mono specific test trees now, or are they all just subsets of the runtime tests in src/tests?

File: docs/workflow/testing/libraries/filtering-tests.md [hunk 5]
```diff
@@ -1,8 +1,18 @@
...
+
+`TestRuntimes` (CoreCLR, Mono) is defined [here](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/TestRuntimes.cs).
+
+`TargetFrameworkMonikers` (Netcoreapp, NetFramework) is defined [here](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/TargetFrameworkMonikers.cs).
+
+These attributes are specified above the test method's definition. The available attributes are:
+
+## OuterLoopAttribute
 
 ```cs
```
Comment (danmoseley) [inline docs/workflow/testing/libraries/filtering-tests.md:17]: another outerloop reason is that the test is disruptive, eg., it pops up a dialog.

we have been de-facto using outerloop for tests that are a bit less reliable, although that practice is dubious now that we are driving outerloop clean.

File: docs/workflow/testing/libraries/filtering-tests.md [hunk 6]
```diff
@@ -176,3 +234,67 @@ dotnet build <csproj_file> /t:Test /p:TargetOS=windows
...
+
+**Disable for combinations of options:**
+
+There are additional attribute signatures for combinations of these configurations, where all the conditions must be met:
+```cs
+SkipOnCoreClr(string reason, RuntimeConfiguration runtimeConfigurations, RuntimeTestModes testModes)
+SkipOnCoreClr(string reason, TestPlatforms testPlatforms, RuntimeConfiguration runtimeConfigurations)
+SkipOnCoreClr(string reason, TestPlatforms testPlatforms, RuntimeTestModes testMode)
+SkipOnCoreClr(string reason, TestPlatforms testPlatforms, RuntimeConfiguration runtimeConfigurations, RuntimeTestModes testModes)
+```
```
Comment (danmoseley) [inline docs/workflow/testing/libraries/filtering-tests.md:300]: it's probably out of scope, but nowhere better to put it -- `[Collection(nameof(DisableParallelization))]` anywhere. We put it on test classes to indicate that none of the tests in that class (which as usual run serially with respect to each other) may run concurrently with tests in another class. We use this for tests that use a lot of disk space or memory, or dominate all the cores, such that they are likely to disrupt any tests that run concurrently.

File: docs/workflow/testing/libraries/filtering-tests.md (hunk 2 above)
Comment (danmoseley) [reply docs/workflow/testing/libraries/filtering-tests.md:239]: ideally, but there are some filters possible on them that aren't possible on ActiveIssue.

**Review by danmoseley — APPROVED** [review]
thanks for creating these

File: docs/workflow/testing/libraries/filtering-tests.md (hunk 3 above)
Comment (danmoseley) [reply docs/workflow/testing/libraries/filtering-tests.md:265]: Ah yes you can have multiple such attributes

        [SkipOnCoreClr("https://github.com/dotnet/runtime/issues/67886", ~RuntimeConfiguration.Release)]
        [SkipOnCoreClr("https://github.com/dotnet/runtime/issues/67886", RuntimeTestModes.JitMinOpts)]

File: docs/workflow/ci/disabling-tests.md (hunk 4 above)
Comment (BruceForstall) [reply docs/workflow/ci/disabling-tests.md:15]: IIUC, it is the same test tree, but I don't know if Mono has something in addition.

**Review by ViktorHofer — COMMENTED** [review]
LGTM, thanks.

Ideally we would move the content from "filtering-tests.md" into the XunitExtensions arcade project as a README.md, include that file into the package and link to it from here.

**Pull request merged**